variants/Generic_F446Rx/variant.cpp

```diff
@@ -0,0 +1,238 @@
+  PB_8,  // Digital pin 24
+  PB_9,  // Digital pin 25
+  PB_10, // Digital pin 26
+  PB_12, // Digital pin 27
+  PB_13, // Digital pin 28
+  PB_14, // Digital pin 29
+  PB_15, // Digital pin 30
+
+  PC_0,  // Digital pin 31
+  PC_1,  // Digital pin 32
+  PC_2,  // Digital pin 33
+  PC_3,  // Digital pin 34
+  PC_4,  // Digital pin 35
+  PC_5,  // Digital pin 36
+  PC_6,  // Digital pin 37
+  PC_7,  // Digital pin 38
+  PC_8,  // Digital pin 39
+  PC_9,  // Digital pin 40
+  PC_10, // Digital pin 41
+  PC_11, // Digital pin 42
+  PC_12, // Digital pin 43
+  PC_13, // Digital pin 44
+  PC_14, // Digital pin 45
+  PC_15, // Digital pin 46
+
+  PD_2,  // Digital pin 47
+
+  PH_0,  // Digital pin 48, used by the external oscillator
+  PH_1   // Digital pin 49, used by the external oscillator
+};
+
+// Analog (Ax) pin number array
+const uint32_t analogInputPin[] = {
+  0,  // A0,  PA0
+  1,  // A1,  PA1
+  2,  // A2,  PA2
+  3,  // A3,  PA3
+  4,  // A4,  PA4
+  5,  // A5,  PA5
+  6,  // A6,  PA6
+  7,  // A7,  PA7
+  16, // A8,  PB0
+  17, // A9,  PB1
+  31, // A10, PC0
+  32, // A11, PC1
+  33, // A12, PC2
+  34, // A13, PC3
+  35, // A14, PC4
+  36  // A15, PC5
+};
+
+#ifdef __cplusplus
+}
+#endif
+
+// ----------------------------------------------------------------------------
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * @brief  Configures the System clock source, PLL Multiplier and Divider factors,
+ *               AHB/APBx prescalers and Flash settings
+ * @note   This function should be called only once the RCC clock configuration
+ *         is reset to the default reset state (done in SystemInit() function).
+ * @param  None
+ * @retval None
+ */
+
+/******************************************************************************/
+/*            PLL (clocked by HSE) used as System clock source                */
+/******************************************************************************/
+static uint8_t SetSysClock_PLL_HSE(uint8_t bypass)
+{
+  RCC_OscInitTypeDef RCC_OscInitStruct;
+  RCC_ClkInitTypeDef RCC_ClkInitStruct;
+
+  /* The voltage scaling allows optimizing the power consumption when the device is
+  clocked below the maximum system frequency, to update the voltage scaling value
+  regarding system frequency refer to product datasheet. */
+  __HAL_RCC_PWR_CLK_ENABLE();
+  __HAL_PWR_VOLTAGESCALING_CONFIG(PWR_REGULATOR_VOLTAGE_SCALE2);
+
+  // Enable HSE oscillator and activate PLL with HSE as source
+  RCC_OscInitStruct.OscillatorType      = RCC_OSCILLATORTYPE_HSE;
+  if (bypass == 0) {
+    RCC_OscInitStruct.HSEState          = RCC_HSE_ON; // External 8 MHz xtal on OSC_IN/OSC_OUT
+  } else {
+    RCC_OscInitStruct.HSEState          = RCC_HSE_BYPASS; // External 8 MHz clock on OSC_IN
+  }
+
+  RCC_OscInitStruct.PLL.PLLState        = RCC_PLL_ON;
+  RCC_OscInitStruct.PLL.PLLSource       = RCC_PLLSOURCE_HSE;
+  RCC_OscInitStruct.PLL.PLLM            = 8;             // VCO input clock = 1 MHz (8 MHz / 8)
```

Comment:
As discussed in other issue, the clock config have to use `HSE_VALUE` instead of hardcoded value

Reply:
Sure! So is the solution provided in #880 for HSE/HSI OK?

Reply:
If so, I can update the other pinouts with this fix as well

Reply:
Yes, I've made a comment on this in the dedicated PR

Reply:
Great! I'm a bit busy at the moment, but I'll try to get this all done and tested within this week. OK? 🙂

Reply:
Sure. Thanks